### PR TITLE
docs: Update platform lib README.md with i18n section

### PR DESCRIPTION
**Pull request by kavya-b**
#### Please provide a link to the associated issue.
#2506 

#### Please provide a brief summary of this pull request.
Updates the Readme.md file with a section on i18n support for platform lib.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`

### CODE REVIEW

**Comment by fkolar** [comment]
In terms of example what I would mentions is for user to be carefull to use `FundamentalNgxPlatformModule` import as it contains everything and that we recomend using individual components module what application really needs.

**Comment by kavya-b** [comment]
@sKudum 
> There is mainly 2 kind of people going to access our readme.
> 
> 1. Component Developers(we)==> it has the details for us.
> 2. Customers==> they will be using our components. I did not find details for them.
>    Can we have i18n under developer guidelines and other information in general?

>  Other than this rest looks good.

When you say developer guidelines I am assuming it is this readme itself? Or is there a separate section in this repo or Readme where I should be placing my text instead?
In both 1 and 2 cases, I think it is useful for even customers(developers from customers) to go through the same document in detail. That document itself captures the general use of i18n markers as well. I cannot think of any other general information that might be needed to be separately put for customers. If you could elaborate on that, I can address it. 



@fkolar 
> In terms of example what I would mentions is for user to be carefull to use `FundamentalNgxPlatformModule` import as it contains everything and that we recomend using individual components module what application really needs.

Okay, so after the example after this line 
> To include an individual Angular Fundamental component in your application, you only need to import the relevant module.

I can add the following:
> Be careful while importing the entire library as it loads all modules; we recommend to only import relevant modules as needed.

Is this okay?

**Comment by sKudum** [comment]
> @sKudum
> 
> > There is mainly 2 kind of people going to access our readme.
> > 
> > 1. Component Developers(we)==> it has the details for us.
> > 2. Customers==> they will be using our components. I did not find details for them.
> >    Can we have i18n under developer guidelines and other information in general?
> 
> > Other than this rest looks good.
> 
> When you say developer guidelines I am assuming it is this readme itself? Or is there a separate section in this repo or Readme where I should be placing my text instead?
> In both 1 and 2 cases, I think it is useful for even customers(developers from customers) to go through the same document in detail. That document itself captures the general use of i18n markers as well. I cannot think of any other general information that might be needed to be separately put for customers. If you could elaborate on that, I can address it.
```
@kavya-b 
We have 2 sections in Readme:

1). **Getting Started:**
     this is mainly for customers who are going to use our lib, in this we can add a subsection for i18n and we can share how they can use it. 
for example, an example of a piece of code for imperative and declarative with i18n will do.

2).**Contribution:**

  These for developers who are contributing to our lib here link to platform lib home and the details which are present already are enough.

```

**Comment by kavya-b** [comment]
> > @sKudum
> > > There is mainly 2 kind of people going to access our readme.
> > > 
> > > 1. Component Developers(we)==> it has the details for us.
> > > 2. Customers==> they will be using our components. I did not find details for them.
> > >    Can we have i18n under developer guidelines and other information in general?
> > 
> > 
> > > Other than this rest looks good.
> > 
> > 
> > When you say developer guidelines I am assuming it is this readme itself? Or is there a separate section in this repo or Readme where I should be placing my text instead?
> > In both 1 and 2 cases, I think it is useful for even customers(developers from customers) to go through the same document in detail. That document itself captures the general use of i18n markers as well. I cannot think of any other general information that might be needed to be separately put for customers. If you could elaborate on that, I can address it.
> 
> ```
> @kavya-b 
> We have 2 sections in Readme:
> 
> 1). **Getting Started:**
>      this is mainly for customers who are going to use our lib, in this we can add a subsection for i18n and we can share how they can use it. 
> for example, an example of a piece of code for imperative and declarative with i18n will do.
> 
> 2).**Contribution:**
> 
>   These for developers who are contributing to our lib here link to platform lib home and the details which are present already are enough.
> ```

I see. However, I don't think putting it in this readme.md file makes much sense since it is only talking about basic usage of component. But I think https://github.com/SAP/fundamental-ngx/blob/master/PLATFORM_README.md this document(which I think also ends up in the documentation pages' home page of platform) can be the place for adding an i18n example snippet. Shall I add it there instead?

**Comment by sKudum** [comment]
> > > @sKudum
> > > > There is mainly 2 kind of people going to access our readme.
> > > > 
> > > > 1. Component Developers(we)==> it has the details for us.
> > > > 2. Customers==> they will be using our components. I did not find details for them.
> > > >    Can we have i18n under developer guidelines and other information in general?
> > > 
> > > 
> > > > Other than this rest looks good.
> > > 
> > > 
> > > When you say developer guidelines I am assuming it is this readme itself? Or is there a separate section in this repo or Readme where I should be placing my text instead?
> > > In both 1 and 2 cases, I think it is useful for even customers(developers from customers) to go through the same document in detail. That document itself captures the general use of i18n markers as well. I cannot think of any other general information that might be needed to be separately put for customers. If you could elaborate on that, I can address it.
> > 
> > 
> > ```
> > @kavya-b 
> > We have 2 sections in Readme:
> > 
> > 1). **Getting Started:**
> >      this is mainly for customers who are going to use our lib, in this we can add a subsection for i18n and we can share how they can use it. 
> > for example, an example of a piece of code for imperative and declarative with i18n will do.
> > 
> > 2).**Contribution:**
> > 
> >   These for developers who are contributing to our lib here link to platform lib home and the details which are present already are enough.
> > ```
> 
> I see. However, I don't think putting it in this readme.md file makes much sense since it is only talking about basic usage of component. But I think https://github.com/SAP/fundamental-ngx/blob/master/PLATFORM_README.md this document(which I think also ends up in the documentation pages' home page of platform) can be the place for adding an i18n example snippet. Shall I add it there instead?

Getting started and contibution is in sync for readme.md and PLATFORM_README.md

**Comment by fkolar** [comment]
> @sKudum
> 
> > There is mainly 2 kind of people going to access our readme.
> > 
> > 1. Component Developers(we)==> it has the details for us.
> > 2. Customers==> they will be using our components. I did not find details for them.
> >    Can we have i18n under developer guidelines and other information in general?
> 
> > Other than this rest looks good.
> 
> When you say developer guidelines I am assuming it is this readme itself? Or is there a separate section in this repo or Readme where I should be placing my text instead?
> In both 1 and 2 cases, I think it is useful for even customers(developers from customers) to go through the same document in detail. That document itself captures the general use of i18n markers as well. I cannot think of any other general information that might be needed to be separately put for customers. If you could elaborate on that, I can address it.
> 
> @fkolar
> 
> > In terms of example what I would mentions is for user to be carefull to use `FundamentalNgxPlatformModule` import as it contains everything and that we recomend using individual components module what application really needs.
> 
> Okay, so after the example after this line
> 
> > To include an individual Angular Fundamental component in your application, you only need to import the relevant module.
> 
> I can add the following:
> 
> > Be careful while importing the entire library as it loads all modules; we recommend to only import relevant modules as needed.
> 
> Is this okay?

More Be careful while importing the entire PlatformModule as it loads all modules; we recommend to only import relevant modules as needed.

**Comment by kavya-b** [comment]
> > > > @sKudum
> > > > > There is mainly 2 kind of people going to access our readme.
> > > > > 
> > > > > 1. Component Developers(we)==> it has the details for us.
> > > > > 2. Customers==> they will be using our components. I did not find details for them.
> > > > >    Can we have i18n under developer guidelines and other information in general?
> > > > 
> > > > 
> > > > > Other than this rest looks good.
> > > > 
> > > > 
> > > > When you say developer guidelines I am assuming it is this readme itself? Or is there a separate section in this repo or Readme where I should be placing my text instead?
> > > > In both 1 and 2 cases, I think it is useful for even customers(developers from customers) to go through the same document in detail. That document itself captures the general use of i18n markers as well. I cannot think of any other general information that might be needed to be separately put for customers. If you could elaborate on that, I can address it.
> > > 
> > > 
> > > ```
> > > @kavya-b 
> > > We have 2 sections in Readme:
> > > 
> > > 1). **Getting Started:**
> > >      this is mainly for customers who are going to use our lib, in this we can add a subsection for i18n and we can share how they can use it. 
> > > for example, an example of a piece of code for imperative and declarative with i18n will do.
> > > 
> > > 2).**Contribution:**
> > > 
> > >   These for developers who are contributing to our lib here link to platform lib home and the details which are present already are enough.
> > > ```
> > 
> > 
> > I see. However, I don't think putting it in this readme.md file makes much sense since it is only talking about basic usage of component. But I think https://github.com/SAP/fundamental-ngx/blob/master/PLATFORM_README.md this document(which I think also ends up in the documentation pages' home page of platform) can be the place for adding an i18n example snippet. Shall I add it there instead?
> 
> Getting started and contibution is in sync for readme.md and PLATFORM_README.md

Correct me if I am wrong, but https://github.com/SAP/fundamental-ngx/tree/master/libs/platform this reads README.md and https://sap.github.io/fundamental-ngx/#/platform/home reads PLATFORM_README.md, and these are different.

**Comment by kavya-b** [comment]
@fkolar @sKudum have addressed the comments in latest commit. Please review.

File: PLATFORM_README.md [hunk 1]
```diff
@@ -127,24 +127,49 @@ For an existing Angular CLI application,
...
+    For example, to translate the button text, add `i18n` on the element `fdp-button`:
+
+    ```html
+    <fdp-button i18n="@@emphasizedBtn" [type]="'emphasized'">Emphasized Button</fdp-button>
     ```
 
-3. **Add the component to your HTML.**
+    It is recommended to give ID to the marker like `@@emphasizedBtn` for easier tracking in the extracted `messages.xlf` file.
+
+    Note: Some components allow you to provide data imperatively or through an array of objects like `[list]="seasons"` in the below example:
```
Comment (KevinOkamoto) [inline PLATFORM_README.md:158]: I would remove the "or" in this sentence:
> Some components allow you to provide data imperatively ~~or~~ through an array of objects like `[list]="seasons"` in the below example

File: PLATFORM_README.md [hunk 2]
```diff
@@ -127,24 +127,49 @@ For an existing Angular CLI application,
...
+        [id]="'radio1'"
+        [name]="'radio1'" 
+        [list]="seasons" 
+        [value]="'Summer'" 
+        formControlName="example1">
+    </fdp-radio-group>
     ```
 
+    When data is provided like this(probably coming from a db or backend server), it is left to the application to provide for the translated
+    text themselves. Providing i18n markers as shown in the previous example will not work as this is a limitation of Angular's i18n for now.
```
Comment (KevinOkamoto) [inline PLATFORM_README.md:171]: Add a space before the first parenthesis "(" in this sentence, and I think you can remove the word "themselves".
> When data is provided like this (probably coming from a db or backend server), it is left to the application to provide for the translated
>   text ~~themselves~~.

File: PLATFORM_README.md (hunk 1 above)
Comment (kavya-b) [reply PLATFORM_README.md]: done

File: PLATFORM_README.md (hunk 2 above)
Comment (kavya-b) [reply PLATFORM_README.md]: good catch; done.

File: libs/platform/README.md [hunk 3]
```diff
@@ -82,18 +81,18 @@ For an existing Angular CLI application,
     import { ToggleModule } from '@fundamental-ngx/core';
     ```
 
+    Note: Be careful while importing the entire `FundamentalNgxPlatformModule` as it loads all modules; we recommend to only import relevant modules as needed.
+
     For models prior to 0.11.1 use `fundamental-ngx`
 
+    ```javascript
```
Comment (droshev) [inline libs/platform/README.md:88]: does this make sense? Maybe you wanna give an example with a platform component instead.

File: PLATFORM_README.md [hunk 4]
```diff
@@ -127,24 +127,44 @@ For an existing Angular CLI application,
...
     ...
     imports: [SwitchModule],
     })
     export class DemoModule { }
+    ```
+
+3. **Add the component to your HTML.**
 
+    ```html
+    <fd-switch [size]="'l'" [(checked)]="myValue">Large Switch</fd-switch>
```
Comment (droshev) [inline PLATFORM_README.md:143]: do you want to add a platform example?

File: libs/platform/README.md (hunk 3 above)
Comment (kavya-b) [reply libs/platform/README.md:88]: have added a platform component example instead.

File: PLATFORM_README.md (hunk 4 above)
Comment (kavya-b) [reply PLATFORM_README.md]: have added a platform component example instead.

**Comment by kavya-b** [comment]
@fkolar please approve this PR so that it can be merged. Thanks.

**Pull request merged**